modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/index/SqlPartitionEvictionTest.java

```diff
@@ -0,0 +1,210 @@
+import org.apache.ignite.internal.processors.cache.persistence.CacheDataRow;
+import org.apache.ignite.internal.processors.query.GridQueryProcessor;
+import org.apache.ignite.internal.processors.query.GridQueryTypeDescriptor;
+import org.apache.ignite.internal.processors.query.h2.IgniteH2Indexing;
+import org.apache.ignite.internal.util.typedef.G;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+/** */
+@RunWith(Parameterized.class)
+public class SqlPartitionEvictionTest extends GridCommonAbstractTest {
+    /** */
+    private static final String POI_CACHE_NAME = "POI_CACHE";
+
+    /** */
+    private static final String POI_SCHEMA_NAME = "DOMAIN";
+
+    /** */
+    private static final String POI_TABLE_NAME = "POI";
+
+    /** */
+    private static final String POI_CLASS_NAME = "PointOfInterest";
+
+    /** */
+    private static final String ID_FIELD_NAME = "id";
+
+    /** */
+    private static final String NAME_FIELD_NAME = "name";
+
+    /** */
+    private static final String LATITUDE_FIELD_NAME = "latitude";
+
+    /** */
+    private static final String LONGITUDE_FIELD_NAME = "longitude";
+
+    /** */
+    private static final int NUM_ENTITIES = 1_000;
+
+    /** Test parameters. */
+    @Parameterized.Parameters(name = "backups_count={0}")
+    public static Iterable<Object[]> params() {
+        return Arrays.asList(
+                new Object[] { 0 },
+                new Object[] { 1 },
+                new Object[] { 2 }
+            );
+    }
+
+    /** */
+    @Parameterized.Parameter
+    public int backupsCount;
+
+    private static CountDownLatch latch;
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        cfg.setCacheConfiguration(new CacheConfiguration<>(POI_CACHE_NAME)
+            .setAtomicityMode(CacheAtomicityMode.ATOMIC)
+            .setSqlSchema("DOMAIN")
+            .setQueryEntities(Collections.singletonList(queryEntity()))
+            .setWriteSynchronizationMode(CacheWriteSynchronizationMode.FULL_SYNC)
+            .setCacheMode(CacheMode.PARTITIONED)
+            .setBackups(backupsCount)
+        );
+
+        cfg.setActiveOnStart(true);
+
+        return cfg;
+    }
+
+    @Override protected void beforeTest() throws Exception {
+        latch = new CountDownLatch(1);
+
+        super.beforeTest();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        stopAllGrids(true);
+
+        super.afterTest();
+    }
+
+    @Test
+    public void testSqlConsistencyOnEviction() throws Exception {
+        IgniteEx ig = null;
+        int idx = 0;
```

Comment:
May be just
```
IgniteEx ig = null;

 for(int idx = 0; idx <= backupsCount; idx++)
        ig = ignitionStart(idx)
```

Reply:
But I'm not inist, just extra empty line as separator is enough (after ig definition)